modin/pandas/dataframe.py

```diff
@@ -36,12 +36,15 @@
 from .iterator import PartitionIterator
 from .series import Series
 from .base import BasePandasDataset
+from .groupby import DataFrameGroupBy
+
+from modin.experimental.cloud.meta_magic import make_wrapped_class
 
 
 @_inherit_docstrings(
     pandas.DataFrame, excluded=[pandas.DataFrame, pandas.DataFrame.__init__]
 )
-class DataFrame(BasePandasDataset):
+class _DataFrame(BasePandasDataset):
```

Comment:
This is a pretty major change. Is monkey patching not possible?

Reply:
Have you tried such a concept? `DataFrame.__new__` is a factory method and, depending on the context, returns a remote or local dataframe. To successfully execute methods of a remote dataframe, it is necessary to transfer local objects to the remote context using `deliver` function, but we also have the ability to get objects from local to remote using `obtain` function. To use it, you most likely we need to define another context that will be in the cluster when remote computing is performed. To manage requests, namely to be able to cache, we need to modify the `netref` class, maybe try to inherit from it, or try to use a special API inside `RPyC` for this(if exist).